paymentsheet/src/main/java/com/stripe/android/link/ui/inline/InlineSignupViewModel.kt

```diff
@@ -340,7 +348,13 @@ internal class InlineSignupViewModel(
             }
             LinkSignupMode.InsteadOfSaveForFutureUse -> {
                 when {
-                    linkSignUpOptInFeatureEnabled -> SignUpConsentAction.SignUpOptInMobileChecked
+                    linkSignUpOptInFeatureEnabled -> {
+                        if (config.linkSignUpOptInInitialValue) {
```

Comment:
This needs to look at whether the user has toggled the checkbox off and then on again.

Reply:
done! https://github.com/stripe/stripe-android/pull/11229/commits/fcce58179e906015b4eb26e2a776dbc4061085f3

Reply:
check also the next commit, we need to keep updating the consent action with toggle changes.